src/jit/lowerxarch.cpp

```diff
@@ -2119,10 +2155,28 @@ void Lowering::TreeNodeInfoInitPutArgStk(GenTreePutArgStk* putArgStk)
             prevOffset = fieldOffset;
         }
 
-        // If all fields of this list are slots, set the copy kind.
+        // Set the copy kind.
+        // TODO-X86-CQ: Even if we are using push, if there are contiguous floating point fields, we should
+        // adjust the stack once for those fields. The latter is really best done in code generation, but
+        // this tuning should probably be undertaken as a whole.
+        // Also, if there are  floating point fields, it may be better to use the "Unroll" mode
+        // of copying the struct as a whole, if the fields are not register candidates.
```

Comment:
Will do. I thought I would wait and see the response from my comments on this latest commit.

---
In reply to: [88320293](https://github.com/dotnet/coreclr/pull/7847#discussion_r88320293) [](ancestors = 88320293)